daemon/containerd/image_exporter.go

```diff
@@ -58,18 +60,28 @@ func (i *ImageService) ExportImage(ctx context.Context, names []string, outStrea
 		archive.WithPlatform(platform),
 	}
 
-	ctx, release, err := i.client.WithLease(ctx)
+	contentStore := i.client.ContentStore()
+	leasesManager := i.client.LeasesService()
+	lease, err := leasesManager.Create(ctx, leases.WithRandomID())
```

Comment:
```suggestion
	lease, err := leasesManager.Create(ctx, leases.WithRandomID(), leases.WithExpiration(1*time.Hour))
```
For paranoid safety's sake, should we create the lease with an expiration (a la https://github.com/moby/moby/blob/0c6b61665668fa68538a630d341ad56e53817162/daemon/containerd/image_commit.go#L71C67-L71C101) ?

Reply:
If we use 1 Hour as a timeout in more places, perhaps it would make sense to define a `const` for it that we can document as "this is what we picked".

Silly question though; is there a relation between `ctx` and "expiration"? i.e., should leases also expire if the context is cancelled or timed out?

(if there's no relation by default (which also makes sense if the least is to be used for multiple contexts / calls), I'm wondering if containerd should have a `WithScopedOnContext(ctx)` kinda option that deletes the lease if the context is done).

Reply:
I think our best bet would be to add a cleanup routine at startup that would remove all dangling leases. This should only happen if there is a catastrophic failure and the daemon was killed mid-export. Adding a 1 hour lease would make exports that run more than 1h fail if someone removed the content while we were exporting. Both of the cases should be rare and yeah, a cleanup routine at startup would be best I think. 

We could even add a check in all our tests checking that there are no dangling leases.

> is there a relation between ctx and "expiration"? i.e., should leases also expire if the context is cancelled or timed out?

There isn't, not in the way you describe it no, would be a nice addition for sure. Containerd has only this:

```go
ctx, release, err := i.client.WithLease(ctx)
```

In this case when `release` is called and the lease would be removed. This lease only works for content that is created during the life of that context. In the case of exporting we aren't creating any content, we are only reading them, that's why we need to pre-lease all the content and _then_ safely read it.

Reply:
> I think our best bet would be to add a cleanup routine at startup that would remove all dangling leases.

I'm okay with this, and sounds like a better way to go than arbitrary 1-hour lease. I'll try to write this down to remember to do this somewhere, and when we do that we should go through the rest of the c8d integration code and remove the arbitrary 1-hour expirations in other places (so that an hour long `docker commit` won't break, even though I hope that's not happening frequently)

Reply:
We should also make sure we always set options to the `WithLease` call, by default it will create a 24h lease.

Reply:
Indeed! Forgot about that one

Reply:
I opened a issue to track this one https://github.com/moby/moby/issues/46020